app/src/main/java/pl/targosz/invoicing/database/file/FileBasedInvoiceRepository.java

```diff
@@ -0,0 +1,91 @@
+package pl.targosz.invoicing.database.file;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import java.io.IOException;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import lombok.AllArgsConstructor;
+import lombok.SneakyThrows;
+import pl.targosz.invoicing.config.FileConfiguration;
+import pl.targosz.invoicing.database.InvoiceRepository;
+import pl.targosz.invoicing.model.Invoice;
+import pl.targosz.invoicing.services.FileService;
+import pl.targosz.invoicing.services.JsonService;
+
+@AllArgsConstructor
+public class FileBasedInvoiceRepository implements InvoiceRepository {
+
+    private final FileService invoicesFileService = new FileService(FileConfiguration.INVOICES_DB_PATH);
+    private final FileService idsFilesService = new FileService(FileConfiguration.ID_DB_PATH);
+    private final JsonService<Invoice> jsonService = new JsonService<>();
+
+    @Override
+    public Invoice save(Invoice invoice) {
+        invoice.setId(UUID.randomUUID());
+        try {
+            String json = jsonService.toJson(invoice);
+            invoicesFileService.writeToFile(json);
+            idsFilesService.writeToFile(invoice.getId().toString());
+            return invoice;
+        } catch (IOException e) {
+            throw new IllegalArgumentException();
+        }
+    }
+
+    @Override
+    public Optional<Invoice> getById(UUID id) throws IOException {
+        return getAll().stream().filter(invoice -> invoice.getId().equals(id)).findFirst();
+
+    }
+
+    @Override
+    public List<Invoice> getAll() throws IOException {
+        return invoicesFileService.readFile().map(
+            item -> {
+                try {
+                    return jsonService.toObject(item, Invoice.class);
+                } catch (JsonProcessingException e) {
+                    e.printStackTrace();
```

Comment:
Try to not use e.printStackTrace() - there are better solutions to handle exceptions

Reply:
Done, if not correct, what is the other way?